File: .sdk/tm/php/utility/PrepareMethod.php
<?php
declare(strict_types=1);

// Magic8Ball SDK utility: prepare_method

class Magic8BallPrepareMethod
{
    private const METHOD_MAP = [
        'create' => 'POST',
        'update' => 'PUT',
        'load' => 'GET',
        'list' => 'GET',
        'remove' => 'DELETE',
        'patch' => 'PATCH',
    ];

    public static function call(Magic8BallContext $ctx): string
    {
        return self::METHOD_MAP[$ctx->op->name] ?? 'GET';
    }
}

File: .sdk/tm/rb/utility/prepare_method.rb
# Magic8Ball SDK utility: prepare_method
module Magic8BallUtilities
  METHOD_MAP = { "create"=>"POST", "update"=>"PUT", "load"=>"GET", "list"=>"GET", "remove"=>"DELETE", "patch"=>"PATCH" }
  PrepareMethod = ->(ctx) { METHOD_MAP[ctx.op.name] || "GET" }
end
